### PR TITLE
[extractor/rutube] Support private videos

### PR DESCRIPTION
**Pull request by mexus**
### Add support for "private" videos on Rutube

Rutube requires a special URL argument for private (unlisted) videos (`?p=...`) which needs then to be passed to the API. It also adds a `/private/` fragment to the URL, like the following:

* an unlisted video: `https://rutube.ru/video/private/884fb55f07a97ab673c7d654553e0f48/?p=x2QojCumHTS3rsKHWXN8Lg`
* a normal one: `http://rutube.ru/video/3eac3b4561676c17df9132a9a1e62e3e/`

This PR adds the optional `private` fragment to the URL regexp and for the "secret" argument `?p=...`, and passes it down to the `_download_...` functions where needed.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/rutube.py [hunk 1]
```diff
@@ -16,12 +16,16 @@
 
 
 class RutubeBaseIE(InfoExtractor):
-    def _download_api_info(self, video_id, query=None):
+    def _download_api_info(self, video_id, query=None, secret=None):
```
Comment (pukkandan) [inline yt_dlp/extractor/rutube.py:19]: secret should just be passed as part of `query`. Same below

File: yt_dlp/extractor/rutube.py [hunk 2]
```diff
@@ -129,8 +159,9 @@ def suitable(cls, url):
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
-        info = self._download_and_extract_info(video_id)
-        info['formats'] = self._download_and_extract_formats(video_id)
+        secret = self._match_valid_url(url).group('secret')
```
Comment (pukkandan) [inline yt_dlp/extractor/rutube.py:162]: Isn't it better to pass on all query?
```suggestion
        query = parse_qs(url)
```
or atleast, parse the secret like this - I see no reason to parse it with regex
```suggestion
        query = {'p': parse_qs(url).get('p')}
```
(`from utils import parse_qs`)

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/rutube.py (hunk 2 above)
Comment (mexus) [reply yt_dlp/extractor/rutube.py]: Thanks!! It was so stupid of me to actually parse it with a regex. No the code looks prettier and the changeset has become even smaller.

File: yt_dlp/extractor/rutube.py (hunk 1 above)
Comment (mexus) [reply yt_dlp/extractor/rutube.py]: Yep, thanks!

**Comment by mexus** [comment]
@pukkandan thanks for such a swift review&merge! :)

**Pull request merged**